keptn-gateway/internal/gateway/gateway.go

```diff
@@ -0,0 +1,69 @@
+package gateway
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	_ "github.com/joho/godotenv/autoload"
+	"go.uber.org/zap"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+type Gateway struct {
+	port   int
+	logger *zap.Logger
+	client *kubernetes.Clientset
+}
+
+func NewGateway() *http.Server {
+	port, _ := strconv.Atoi(os.Getenv("PORT"))
+	if port == 0 {
+		port = 8080
+	}
+
+	NewGateway := &Gateway{
```

Comment:
this should also be the struct returned
```suggestion
	gateway := &Gateway{
```